File: python/258_add_digits.py
def addDigits(num: int) -> int:
    num = sum([int(digit) for digit in str(num)])
    if len(str(num)) > 1: num = addDigits(num)
    return num

def addDigitsSmart(num: int) -> int:
    while num > 9: num = num % 10 + num // 10
    return num

File: javascript/258_add_digits.js
/**
 * @param {number} num
 * @return {number}
 */
function addDigits(num) {
    num = num.toString().split("").reduce((a, b) => parseInt(a) + parseInt(b));
    if (num > 9) return addDigits(num);
    return num;
};


/**
 * @param {number} num
 * @return {number}
 */
function addDigitsSmart(num) {
    while (num > 9) {
        num = num % 10 + Math.floor(num / 10);
    }
    return num;
};
